src/lib/abacus/logger.ts

```diff
@@ -11,21 +13,26 @@ class AbacusLogger implements Omit<AbacusLoggingProtocol, '__doNotUseOrImplement
     }
   }
 
-  e(tag: string, message: string) {
+  e(tag: string, message: string, context: object, error: Error) {
     if (
       import.meta.env.VITE_ENABLE_ABACUS_LOGGING ||
       import.meta.env.VITE_ABACUS_LOG_LEVEL === 'error'
     ) {
       console.error(`${tag}: ${message}`);
     }
+    dd.error(message, context, error);
   }
 
-  ddInfo(tag: string, message: string, context?: {}) {
-    if (
-      import.meta.env.VITE_ENABLE_ABACUS_LOGGING ||
-      import.meta.env.VITE_ABACUS_LOG_LEVEL === 'info'
-    ) {
-      console.info(`${tag}: ${message}: ${context}`);
+  ddInfo(tag: string, message: string, context: string) {
+    try {
+      const parsedContext = JSON.parse(context);
+      if (import.meta.env.VITE_DATADOG_LOG_LEVEL === 'debug') {
+        console.log(`${tag} dd info: ${message}`, parsedContext);
+      }
+      dd.info(message, parsedContext);
+      // catch in case parsing context fails for some reason
+    } catch (err) {
+      dd.error('Error sending dd info', undefined, err);
```

Comment:
maybe send the context string in this case?

Reply:
ah good point, forgot the error won't have that data. i like it